include/capo/instance.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+#include <capo/sound.hpp>
+#include <capo/source.hpp>
+#include <capo/types.hpp>
+#include <capo/utils/erased_ptr.hpp>
+#include <capo/utils/id.hpp>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace capo {
+struct PCM;
+
+class Instance {
+  public:
+	Instance();
```

Comment:
Can we have `Instance::create` instead so we can return an `Result` error if the creation fails?

Reply:
After trying this I realized it will be more work, we would need one of the following:
- `Instance` is not RAII (destructor gets called when static function returns)
- it has move semantics but only for this static factory function, can't move after that; also conditional destruction

I think it's fine to report an error via the customization point and disable the instance.